runtime/tests/integration_test.rs

```diff
@@ -0,0 +1,303 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Moonbeam Runtime Integration Tests
+
+#![cfg(test)]
+
+use cumulus_primitives_parachain_inherent::ParachainInherentData;
+use frame_support::{
+	assert_noop, assert_ok,
+	dispatch::Dispatchable,
+	traits::{GenesisBuild, OnFinalize, OnInitialize},
+};
+use moonbeam_runtime::{
+	AccountId, AuthorInherent, Balance, Balances, Call, Event, InflationInfo, ParachainStaking,
+	Range, Runtime, System, GLMR,
+};
+use sp_runtime::{DispatchError, Perbill};
+
+fn run_to_block(n: u32) {
+	while System::block_number() < n {
+		AuthorInherent::on_finalize(System::block_number());
+		ParachainStaking::on_finalize(System::block_number());
+		System::set_block_number(System::block_number() + 1);
+		AuthorInherent::on_initialize(System::block_number());
+	}
+}
+
+fn last_event() -> Event {
+	System::events().pop().expect("Event expected").event
+}
+
+struct ExtBuilder {
+	// balances
+	balances: Vec<(AccountId, Balance)>,
+	// [collator, amount]
+	collators: Vec<(AccountId, Balance)>,
+	// [nominator, collator, nomination_amount]
+	nominators: Vec<(AccountId, AccountId, Balance)>,
+}
+
+impl Default for ExtBuilder {
+	fn default() -> ExtBuilder {
+		ExtBuilder {
+			balances: vec![],
+			nominators: vec![],
+			collators: vec![],
+		}
+	}
+}
+
+impl ExtBuilder {
+	fn with_balances(mut self, balances: Vec<(AccountId, Balance)>) -> Self {
+		self.balances = balances;
+		self
+	}
+
+	fn with_collators(mut self, collators: Vec<(AccountId, Balance)>) -> Self {
```

Comment:
Joshy showed me this, definitely improves readability and I will be making the same change to the parachain-staking tests

Reply:
done